In [100]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Closed_deals

In [108]:
# carga del dataset y muestreo de 10 registros aleatorios
Closed_deals = pd.read_csv('Datasets/olist_closed_deals_dataset.csv')
Closed_deals.sample(10)

mql_id                         seller_id  \
55   3a91a097116d6c8b32d41e0c81dd1e9c  e433f5fd4050e3352b5d83522b7fe24b   
456  fca7ac68a9bcfe7ec3a017257471f198  48fa294ea8e32148c8dba7a692b72bfc   
308  5f7526ec1bdd6eebdf4f7d0054cfe163  bc8c8d665ec4664d286be0d521722b19   
358  e633a3d4fb3db2c7665c170c80db6717  c1319b7d63c25ad7838d268fec93d0fa   
685  61681a30d47dc8b1ec5180ccdec26ba5  4a4da369ad50f14d48337aa52eb826f5   
393  b83a52c035258ed43d787106b1765e35  99a6380f4dd5ed504cd8003d94d0be25   
692  13c5dfa246046c0beea97e100f32a1a9  968268a686aa05d1f529fb07dff08130   
673  8ac5c69653a6ec0c8a96f304c13fb1d6  e58a5b390e28abc0b216cfb0e07d27d7   
778  c2102c66aa81e5c07f72cd88e46d091d  649d307ba392e2cc6cedc7888f51923d   
513  1dbafa1bd602541ba5249cf8cc388acf  252b493131fc46ee4a19fbfcbd565676   

                               sdr_id                             sr_id  \
55   a8387c01a09e99ce014107505b92388c  9e4d1098a3b0f5da39b0bc48f9876645   
456  9e4d1098a3b0f5da39b0bc48f9876645  6565aa9ce3178a5caf6171827af3a9ba   
308  4b339f9567d060bcea4f5136b9f5949e  6565aa9ce3178a5caf6171827af3a9ba   
358  56bf83c4bb35763a51c2baab501b4c67  de63de0d10a6012430098db33c679b0b   
685  068066e24f0c643eb1d089c7dd20cd73  6565aa9ce3178a5caf6171827af3a9ba   
393  b90f87164b5f8c2cfa5c8572834dbe3f  9ae085775a198122c5586fa830ff7f2b   
692  fdb16d3cbbeb5798f2f66c4096be026d  9ae085775a198122c5586fa830ff7f2b   
673  068066e24f0c643eb1d089c7dd20cd73  495d4e95a8cf8bbf8b432b612a2aa328   
778  f42a2bd194f7802ab052a815c8de65b7  2695de1affa7750089c0455f8ce27021   
513  0a0fb2b07d841f84fb6714e35c723075  4ef15afb4b2723d8f3d81e51ec7afefe   

                won_date                 business_segment      lead_type  \
55   2018-04-26 03:00:00  construction_tools_house_garden  online_medium   
456  2018-04-12 12:35:49                   sports_leisure       industry   
308  2018-04-17 17:18:12                       food_drink  online_medium   
358  2018-04-30 21:48:02          audio_video_electronics     online_big   
685  2018-01-29 15:09:03                    health_beauty  online_medium   
393  2018-08-08 21:14:25                       home_decor            NaN   
692  2018-01-22 19:14:25                       stationery     online_big   
673  2018-01-19 19:29:55                   bags_backpacks  online_medium   
778  2018-05-14 14:24:45                  car_accessories        offline   
513  2018-06-04 12:47:47                        computers  online_medium   

    lead_behaviour_profile has_company has_gtin average_stock business_type  \
55                     cat         NaN      NaN           NaN  manufacturer   
456                    cat         NaN      NaN           NaN      reseller   
308                   wolf         NaN      NaN           NaN      reseller   
358                    cat         NaN      NaN           NaN  manufacturer   
685                    NaN         NaN      NaN           NaN      reseller   
393                    NaN         NaN      NaN           NaN      reseller   
692                    NaN         NaN      NaN           NaN      reseller   
673                    NaN         NaN      NaN           NaN      reseller   
778                    cat         NaN      NaN           NaN      reseller   
513                    cat         NaN      NaN           NaN      reseller   

     declared_product_catalog_size  declared_monthly_revenue  
55                             NaN                       0.0  
456                            NaN                       0.0  
308                            NaN                       0.0  
358                            NaN                       0.0  
685                            NaN                       0.0  
393                            NaN                       0.0  
692                            NaN                       0.0  
673                            NaN                       0.0  
778                            NaN                       0.0  
513                            NaN        

In [61]:
# mombres de columnas
Closed_deals.columns

Index(['mql_id', 'seller_id', 'sdr_id', 'sr_id', 'won_date',
       'business_segment', 'lead_type', 'lead_behaviour_profile',
       'has_company', 'has_gtin', 'average_stock', 'business_type',
       'declared_product_catalog_size', 'declared_monthly_revenue'],
      dtype='object')

In [62]:
# tamaño del dataset
Closed_deals.shape

(842, 14)

In [5]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Closed_deals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 842 entries, 0 to 841
Data columns (total 14 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   mql_id                         842 non-null    object 
 1   seller_id                      842 non-null    object 
 2   sdr_id                         842 non-null    object 
 3   sr_id                          842 non-null    object 
 4   won_date                       842 non-null    object 
 5   business_segment               841 non-null    object 
 6   lead_type                      836 non-null    object 
 7   lead_behaviour_profile         665 non-null    object 
 8   has_company                    63 non-null     object 
 9   has_gtin                       64 non-null     object 
 10  average_stock                  66 non-null     object 
 11  business_type                  832 non-null    object 
 12  declared_product_catalog_size  69 non-null     flo

In [6]:
# descripcion de las columnas numéricas
Closed_deals.describe()

declared_product_catalog_size  declared_monthly_revenue
count                      69.000000              8.420000e+02
mean                      233.028986              7.337768e+04
std                       352.380558              1.744799e+06
min                         1.000000              0.000000e+00
25%                        30.000000              0.000000e+00
50%                       100.000000              0.000000e+00
75%                       300.000000              0.000000e+00
max                      2000.000000              5.000000e+07

In [65]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Closed_deals.describe(include=['O'])

mql_id                         seller_id  \
count                                842                               842   
unique                               842                               842   
top     5420aad7fec3549a85876ba1c529bd84  2c43fb513632d29b3b58df74816f1b06   
freq                                   1                                 1   

                                  sdr_id                             sr_id  \
count                                842                               842   
unique                                32                                22   
top     4b339f9567d060bcea4f5136b9f5949e  4ef15afb4b2723d8f3d81e51ec7afefe   
freq                                 140                               133   

                   won_date business_segment      lead_type  \
count                   842              841            836   
unique                  824               33              8   
top     2018-05-04 03:00:00       home_decor  online_medium   
freq                      6              105            332   

       lead_behaviour_profile has_company has_gtin average_stock business_type  
count                     665          63       64            66           832  
unique                      9           2        2             6             3  
top                       cat        True     True          5-20      reseller  
freq                      407          58       54            22           587

In [7]:
# cuantos datos nulos por columna se encuentran
Closed_deals.isna().sum()

mql_id                             0
seller_id                          0
sdr_id                             0
sr_id                              0
won_date                           0
business_segment                   1
lead_type                          6
lead_behaviour_profile           177
has_company                      779
has_gtin                         778
average_stock                    776
business_type                     10
declared_product_catalog_size    773
declared_monthly_revenue           0
dtype: int64

- De un total de 842 registros las columnas has_company, has_gtin, average_stock y declared_product_catalog_size tiene mas del 90 % de sus datos faltantes

## Customers

In [9]:
# carga del dataset y muestreo de 10 registros aleatorios
Customers = pd.read_csv('Datasets/olist_customers_dataset.csv')
Customers.sample(10)

customer_id                customer_unique_id  \
42749  a8a08a106f5596b07e4af3742c244771  6be8ba8106013aa253958431e4bea568   
88577  790a05c9cb4a2ed50d88ef313a7e9dfd  ee4fdf2f9e5f4a1b211db070c78b1215   
72197  ce1bcfe3b3c0ee8f1d421a3eb754d9ce  f0a37349de902094650c18655a82b98c   
29236  edd8be2a0d8a9ceb81141bdb7316648b  3629d3f48ea96718f82645e0dc0525ab   
89353  fd3b44dd13dab141361b58172c65b84d  f8cc76040ce7f952d1a4b4ff8af43d61   
81271  5043e611d34f6ad724ca658aa22cc43c  ba16ab3721ffd6b57fc87a7a4f6a4fcf   
70943  e5df9583ddbe3f4f3857d9b667dfa609  acc38a550c42d17125cd25c32a464d26   
29435  5f6b16b2dae42bcbceb3f79dac11c611  f4e1ffa5c2042ce07aed52d6f949183e   
82456  676f55d05b1059070b493c5e9e741917  5cc22576ed6032e9dab0501f28f0c33f   
79105  bbf257c74411e54169f2f91ce1b9aad9  0efae48cb4ba36016622e290fcfc2f9f   

       customer_zip_code_prefix             customer_city customer_state  
42749                     29141                 cariacica             ES  
88577                     37200                    lavras             MG  
72197                     37810                guaranesia             MG  
29236                     35550               itapecerica             MG  
89353                     68515               parauapebas             PA  
81271                     11420                   guaruja             SP  
70943                     45810              porto seguro             BA  
29435                     21235            rio de janeiro             RJ  
82456                     55190  santa cruz do capibaribe             PE  
79105                     45860               canavieiras             BA

In [66]:
# nombres de columnas
Customers.columns

Index(['customer_id', 'customer_unique_id', 'customer_zip_code_prefix',
       'customer_city', 'customer_state'],
      dtype='object')

In [67]:
# tamaño del dataset
Customers.shape

(99441, 5)

In [10]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


In [11]:
# descripcion de las columnas numéricas 
Customers.describe()

customer_zip_code_prefix
count              99441.000000
mean               35137.474583
std                29797.938996
min                 1003.000000
25%                11347.000000
50%                24416.000000
75%                58900.000000
max                99990.000000

In [68]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Customers.describe(include=['O'])

customer_id                customer_unique_id  \
count                              99441                             99441   
unique                             99441                             96096   
top     06b8999e2fba1a1fbc88172c00ba8bc7  8d50f5eadf50201ccdcedfb9e2ac8455   
freq                                   1                                17   

       customer_city customer_state  
count          99441          99441  
unique          4119             27  
top        sao paulo             SP  
freq           15540          41746

- analizando Customers faltan datos de customers_unique_id que no hay referencia de como fueron creados esos codigos

In [12]:
# cuantos datos nulos por solumna se encuentran
Customers.isna().sum()

customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

## Geolocation

In [14]:
# carga del dataset y muestreo de 10 registros aleatorios
Geolocation = pd.read_csv('Datasets/olist_geolocation_dataset.csv')
Geolocation.sample(10)

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
803384                        75170       -16.510622       -49.014202   
560210                        31035       -19.902243       -43.912610   
293888                        13100       -22.914375       -47.032650   
909246                        88240       -27.274407       -48.850315   
293924                        13171       -22.833981       -47.271958   
625835                        37175       -20.940622       -45.829679   
471862                        24220       -22.901771       -43.111901   
619864                        36773       -21.380840       -42.702658   
241684                         9851       -23.709606       -46.593150   
70501                          3986       -23.610504       -46.522846   

             geolocation_city geolocation_state  
803384            goianapolis                GO  
560210         belo horizonte                MG  
293888               campinas                SP  
909246       sao joao batista                SC  
293924                 sumare                SP  
625835               ilicinea                MG  
471862                niteroi                RJ  
619864             cataguases                MG  
241684  são bernardo do campo                SP  
70501               sao paulo                SP

In [69]:
# nombres de columnas
Geolocation.columns

Index(['geolocation_zip_code_prefix', 'geolocation_lat', 'geolocation_lng',
       'geolocation_city', 'geolocation_state'],
      dtype='object')

In [80]:
# tamaño del dataset
Geolocation.shape

(1000163, 5)

In [15]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Geolocation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000163 entries, 0 to 1000162
Data columns (total 5 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   geolocation_zip_code_prefix  1000163 non-null  int64  
 1   geolocation_lat              1000163 non-null  float64
 2   geolocation_lng              1000163 non-null  float64
 3   geolocation_city             1000163 non-null  object 
 4   geolocation_state            1000163 non-null  object 
dtypes: float64(2), int64(1), object(2)
memory usage: 38.2+ MB


In [16]:
# descripcion de las columnas numéricas
Geolocation.describe()

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng
count                 1.000163e+06     1.000163e+06     1.000163e+06
mean                  3.657417e+04    -2.117615e+01    -4.639054e+01
std                   3.054934e+04     5.715866e+00     4.269748e+00
min                   1.001000e+03    -3.660537e+01    -1.014668e+02
25%                   1.107500e+04    -2.360355e+01    -4.857317e+01
50%                   2.653000e+04    -2.291938e+01    -4.663788e+01
75%                   6.350400e+04    -1.997962e+01    -4.376771e+01
max                   9.999000e+04     4.506593e+01     1.211054e+02

In [71]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Geolocation.describe(include=['O'])

geolocation_city geolocation_state
count           1000163           1000163
unique             8011                27
top           sao paulo                SP
freq             135800            404268

In [19]:
# cuantos datos nulos por columna se encuentran
Geolocation.isna().sum()

geolocation_zip_code_prefix    0
geolocation_lat                0
geolocation_lng                0
geolocation_city               0
geolocation_state              0
dtype: int64

## Marketing

In [21]:
# carga del dataset y muestreo de 10 registros aleatorios
Marketing = pd.read_csv('Datasets/olist_marketing_qualified_leads_dataset.csv')
Marketing.sample(10)

mql_id first_contact_date  \
6049  963172122a0c8dad5d8a9cf46f21f5e2         2018-04-04   
5492  95b7cee06276726dffd84adf981f0c65         2018-03-24   
6836  1ad74caec14332594a6e910b3d183e87         2018-04-11   
4714  3c12c84af346626dc2f1b77e52bb301e         2017-08-23   
514   0f7f9b5535079ff1dd10737feb5e1ce3         2017-08-21   
3153  ee488918cd8d11b893067864f9dabddb         2018-03-15   
5926  e07eba70c6ddb8256e6fb638570f0c80         2018-04-29   
1471  aca704179243b072a128766eb67ad4ab         2018-04-19   
5728  72aedcd54a1ac4e361ac47297c415735         2018-01-23   
5991  8a44192374dd613240bf74f1f2498cfc         2018-05-16   

                       landing_page_id          origin  
6049  58326e62183c14b0c03085c33b9fdc44          social  
5492  58326e62183c14b0c03085c33b9fdc44          social  
6836  40dec9f3d5259a3d2dbcdab2114fae47     paid_search  
4714  e492ee5eaf1697716985cc6f33f9cd9b         unknown  
514   3cd2a8301d5395f21cdd3094c9eaa2f4     paid_search  
3153  b76ef37428e6799c421989521c0e5077         unknown  
5926  a7982125ff7aa3b2054c6e44f9d28522           email  
1471  1ceb590cd1e00c7ee95220971f82693d     paid_search  
5728  ce1a65abd0973638f1c887a6efcfa82d  organic_search  
5991  6a110e795dd487f1cf8d7583671987af  organic_search

In [72]:
# mombres de columnas
Marketing.columns

Index(['mql_id', 'first_contact_date', 'landing_page_id', 'origin'], dtype='object')

In [81]:
# tamaño del dataset
Marketing.shape

(8000, 4)

In [22]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   mql_id              8000 non-null   object
 1   first_contact_date  8000 non-null   object
 2   landing_page_id     8000 non-null   object
 3   origin              7940 non-null   object
dtypes: object(4)
memory usage: 250.1+ KB


In [23]:
# descripcion de las columnas numéricas
Marketing.describe()

mql_id first_contact_date  \
count                               8000               8000   
unique                              8000                336   
top     dac32acd4db4c29c230538b72f8dd87d         2018-05-02   
freq                                   1                 93   

                         landing_page_id          origin  
count                               8000            7940  
unique                               495              10  
top     b76ef37428e6799c421989521c0e5077  organic_search  
freq                                 912            2296

In [91]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Marketing.describe(include=['O'])

mql_id first_contact_date  \
count                               8000               8000   
unique                              8000                336   
top     dac32acd4db4c29c230538b72f8dd87d         2018-05-02   
freq                                   1                 93   

                         landing_page_id          origin  
count                               8000            7940  
unique                               495              10  
top     b76ef37428e6799c421989521c0e5077  organic_search  
freq                                 912            2296

In [24]:
# cuantos datos nulos por columna se encuentran
Marketing.isna().sum()

mql_id                 0
first_contact_date     0
landing_page_id        0
origin                60
dtype: int64

## Order_items

In [26]:
# carga del dataset y muestreo de 10 registros aleatorios
Order_items = pd.read_csv('Datasets/olist_order_items_dataset.csv')
Order_items.sample(10)

order_id  order_item_id  \
17273   2793a15df4b2130bcdbe0f125eee15ea              3   
19663   2d28cc7a503b6547e21d671f6a4a44e0              1   
38682   57df8f4c76508fd7b520deeb75e2587b              1   
67777   9af8448f626289733ce1337988c420f1              1   
81377   b8edc6668eb315d1b730348cac5e9520              1   
46501   69aca49c98f05af05bec6fa1f87a3059              1   
111043  fc4b2a0590430c3cef1c11dc737e4517              1   
100243  e34e1db7516bb12cd9985fe66fe2d323              1   
98291   deea4294a344e9365bcb722dca9047d1              1   
41124   5db31af449c668e4ed0b7efb00f802c1              1   

                              product_id                         seller_id  \
17273   71cedfb0fc49b02e2cb77b7ae1fff332  897060da8b9a21f655304d50fd935913   
19663   5aabb9c72866bc1b6aff2fb42852be3b  6d66611d7c44cc30ce351abc49a68421   
38682   7a10781637204d8d10485c71a6108a2e  4869f7a5dfa277a7dca6462dcf3b52b2   
67777   8a6187b2665118d5095f99a25fd7ba7a  4a3ca9315b744ce9f8e9374361493884   
81377   ae52eeb8383b94455f0a08b1f88ad002  f457c46070d02cadd8a68551231220dd   
46501   c848bbf7ec73272cce4df7509a71afde  141440cad9341e553f8401c574a2aa38   
111043  422879e10f46682990de24d770e7f83d  1f50f920176fa81dab994f9023523100   
100243  2827ebfb043a328a13a06ca9165cad0a  1900267e848ceeba8fa32d80c1a5f5a8   
98291   727a46f0b58a9ef165af9f7ebcfebfa9  e9779976487b77c6d4ac45f75ec7afe9   
41124   9afe5faa4fa31a0858bbac1b3821977a  7d76b645482be4a332374e8223836592   

        shipping_limit_date   price  freight_value  
17273   2018-03-13 11:15:39   49.50          12.69  
19663   2018-02-05 15:55:26  159.90          17.37  
38682   2017-12-15 10:11:07  219.90          18.79  
67777   2017-03-31 13:10:19   87.00          14.78  
81377   2018-02-22 16:31:23   69.90          16.93  
46501   2018-07-24 13:30:13  119.80          27.79  
111043  2017-10-02 12:07:21   59.90          13.44  
100243  2018-02-15 18:55:37   69.00          17.73  
98291   2018-05-18 11:13:59   19.49           7.87  
41124   2017-09-11 10:25:27   24.90          15.79

In [73]:
# mombres de columnas
Order_items.columns

Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='object')

In [82]:
# tamaño del dataset
Order_items.shape

(112650, 7)

In [27]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Order_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB


In [28]:
# descripcion de las columnas numéricas
Order_items.describe()

order_item_id          price  freight_value
count  112650.000000  112650.000000  112650.000000
mean        1.197834     120.653739      19.990320
std         0.705124     183.633928      15.806405
min         1.000000       0.850000       0.000000
25%         1.000000      39.900000      13.080000
50%         1.000000      74.990000      16.260000
75%         1.000000     134.900000      21.150000
max        21.000000    6735.000000     409.680000

In [92]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Order_items.describe(include=['O'])

order_id                        product_id  \
count                             112650                            112650   
unique                             98666                             32951   
top     8272b63d03f5f79c56e9e4120aec44ef  aca2eb7d00ea1a7b8ebd4e68314663af   
freq                                  21                               527   

                               seller_id  shipping_limit_date  
count                             112650               112650  
unique                              3095                93318  
top     6560211a19b47992c3666cc44a7e94c0  2017-07-21 18:25:23  
freq                                2033                   21

In [29]:
# cuantos datos nulos por columna se encuentran
Order_items.isna().sum()

order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

## Order_payments

In [31]:
# carga del dataset y muestreo de 10 registros aleatorios
Order_payments = pd.read_csv('Datasets/olist_order_payments_dataset.csv')
Order_payments.sample(10)

order_id  payment_sequential payment_type  \
69378  09bfe605baffbe939fb10d3e8936346e                   1       boleto   
20322  48ccdeef5374f816e8e7112ce99c8076                   1  credit_card   
46827  134383561b50fa8eca9c19b2c792d74c                   1  credit_card   
44409  53a72dbf29b3890aeae332e05c763169                   1  credit_card   
20152  302e27012670f8e2db4418e6c619f54c                   1  credit_card   
85819  d6e2b70c5dd291b6a91fa86497fd9221                   1  credit_card   
1973   70498ad5af42884a122355352f008c14                   2  credit_card   
15834  267e7e0b0615d34fa4c39c0a97cee1d5                   1       boleto   
33926  036bbb637f99f662c928d303d461ffe1                   1  credit_card   
40695  da9aa61ec7bb8713f483bf0bf94c71c2                   2  credit_card   

       payment_installments  payment_value  
69378                     1          43.00  
20322                     2         135.49  
46827                     2         160.76  
44409                     3         190.37  
20152                     3          96.90  
85819                     3          68.32  
1973                      1         239.39  
15834                     1         117.63  
33926                     4         220.37  
40695                     3         107.21

In [74]:
# mombres de columnas
Order_payments.columns

Index(['order_id', 'payment_sequential', 'payment_type',
       'payment_installments', 'payment_value'],
      dtype='object')

In [83]:
# tamaño del dataset
Order_payments.shape

(103886, 5)

In [32]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Order_payments.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


In [33]:
# descripcion de las columnas numéricas
Order_payments.describe()

payment_sequential  payment_installments  payment_value
count       103886.000000         103886.000000  103886.000000
mean             1.092679              2.853349     154.100380
std              0.706584              2.687051     217.494064
min              1.000000              0.000000       0.000000
25%              1.000000              1.000000      56.790000
50%              1.000000              1.000000     100.000000
75%              1.000000              4.000000     171.837500
max             29.000000             24.000000   13664.080000

In [93]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Order_payments.describe(include=['O'])

order_id payment_type
count                             103886       103886
unique                             99440            5
top     fa65dad1b0e818e3ccc5cb0e39231352  credit_card
freq                                  29        76795

In [34]:
# cuantos datos nulos por columna se encuentran
Order_payments.isna().sum()

order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

## Order_reviews

In [36]:
# carga del dataset y muestreo de 10 registros aleatorios
Order_reviews = pd.read_csv('Datasets/olist_order_reviews_dataset.csv')
Order_reviews.sample(10)

review_id                          order_id  \
2603   415bcb1e623a03d4460e7c842b438ca3  202bcf28f7e3dfcee06ab0bfe10c750e   
11048  6134d0def934fe2fe51f2e1753999c39  1def7af24cb7182497acfd953f6dd88b   
64511  eb58451b73d24f7966f51597c67b5429  4fc010de2244ff9ee4e5e334699824bf   
41001  bf960064ef66c8b58583bac6354368af  1c55f7ccb17dea1623c2fc6be250ca6f   
32717  c7c843a11b6a2590f949b58350beef6a  34b426d7bae7fa280beff9a3cda190da   
38666  53719b269cbb2d8b7854550df37c3813  a0c9d84c9ca938bbf711acbd940ba416   
63754  116fe7ea64cdd223ae905d7007e95ae1  12246bc7c8c246d8294f4f6a04efc60e   
59935  d97df97fa17c7743011c15c8c2767283  3a511b11011bb31230c08bc11f4a1954   
79738  fdc5dc724a680827baa4b1a168935a4a  68db1e9c4b7bb26ce60fc4749d8b904d   
78804  67d3361a9a79d737ee39facf61be5e75  f8bbef6f308acd23949b5560c282ede9   

       review_score review_comment_title  \
2603              5                  NaN   
11048             1                  NaN   
64511             5                  NaN   
41001             5                  NaN   
32717             5                  NaN   
38666             5                  NaN   
63754             5                  NaN   
59935             5                  NaN   
79738             5                  NaN   
78804             5                  NaN   

                                  review_comment_message review_creation_date  \
2603                                                 NaN  2018-08-09 00:00:00   
11048                                                NaN  2018-02-24 00:00:00   
64511  Foi rápido a entrega, produto dentro das expec...  2017-05-27 00:00:00   
41001  gostei do produto chegou no prazo certo. Satis...  2017-08-02 00:00:00   
32717                                                NaN  2018-06-30 00:00:00   
38666                                       Muito bom...  2017-10-06 00:00:00   
63754                                                NaN  2017-03-17 00:00:00   
59935                                                NaN  2018-07-24 00:00:00   
79738  Produto entregue antes do prazo.. tudo funcion...  2017-03-25 00:00:00   
78804                                                NaN  2017-07-18 00:00:00   

      review_answer_timestamp  
2603      2018-08-12 00:43:50  
11048     2018-02-26 11:10:17  
64511     2017-05-29 15:11:21  
41001     2017-08-03 01:11:12  
32717     2018-06-30 22:29:15  
38666     2017-10-07 12:59:34  
63754     2017-03-20 10:21:15  
59935     2018-07-25 22:07:42  
79738     2017-03-27 13:22:25  
78804     2017-07-18 23:36:52

In [75]:
# mombres de columnas
Order_reviews.columns

Index(['review_id', 'order_id', 'review_score', 'review_comment_title',
       'review_comment_message', 'review_creation_date',
       'review_answer_timestamp'],
      dtype='object')

In [84]:
# tamaño del dataset
Order_reviews.shape

(99224, 7)

In [37]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Order_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   review_id                99224 non-null  object
 1   order_id                 99224 non-null  object
 2   review_score             99224 non-null  int64 
 3   review_comment_title     11568 non-null  object
 4   review_comment_message   40977 non-null  object
 5   review_creation_date     99224 non-null  object
 6   review_answer_timestamp  99224 non-null  object
dtypes: int64(1), object(6)
memory usage: 5.3+ MB


In [38]:
# descripcion de las columnas numéricas
Order_reviews.describe()

review_score
count  99224.000000
mean       4.086421
std        1.347579
min        1.000000
25%        4.000000
50%        5.000000
75%        5.000000
max        5.000000

In [94]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Order_reviews.describe(include=['O'])

review_id                          order_id  \
count                              99224                             99224   
unique                             98410                             98673   
top     7b606b0d57b078384f0b58eac1d41d78  c88b1d1b157a9999ce368f218a407141   
freq                                   3                                 3   

       review_comment_title review_comment_message review_creation_date  \
count                 11568                  40977                99224   
unique                 4527                  36159                  636   
top               Recomendo              Muito bom  2017-12-19 00:00:00   
freq                    423                    230                  463   

       review_answer_timestamp  
count                    99224  
unique                   98248  
top        2017-06-15 23:21:05  
freq                         4

In [39]:
# cuantos datos nulos por columna se encuentran
Order_reviews.isna().sum()

review_id                      0
order_id                       0
review_score                   0
review_comment_title       87656
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64

## Orders

In [41]:
# carga del dataset y muestreo de 10 registros aleatorios
Orders = pd.read_csv('Datasets/olist_orders_dataset.csv')
Orders.sample(10)

order_id                       customer_id  \
51803  fd74ee83ca49ea03ce7b1e8d8aceba3f  140840d08a638fb3c2427dd91d929fe2   
49025  459c8e5f96858970a0dc1fc5158363d5  0c3a2d52180381fae19478346cfeff76   
46869  11d7140d6b8215c6277987fe827e932b  a5a43c1ea3080ebec9a85b2367a8ceb1   
91077  de2e5a0863ec922ee9fb96b8b3395de2  087f9b562f7f24274e418c71ed7cd56c   
77800  2f371d97b4ea52be4c04ef49bb4f31ff  d782082d67302107f082c112b324aebb   
15320  723e462ce1ee50e024887c0b403130f3  1353b6db7e448ecfad1a4cd13b95f832   
31283  94876636c0c9ddd3455bffe120d95c68  e1d569375b1ae9c7f97e864718ab4b50   
29704  e485751217522068a393e874ed8b5298  e5e221713049b4ca7d373340f611c2bd   
59310  5bc4d0aa53124192dd06058bdf74a6a7  ad1e6270f2b9ecef0068bacaf3357338   
75314  4b9d4dd62814524c9244f8ca0bd377d1  e455f2d619b05659221d21dcb2f1de15   

      order_status order_purchase_timestamp    order_approved_at  \
51803    delivered      2017-06-08 20:31:14  2017-06-08 20:45:19   
49025    delivered      2018-01-04 14:44:23  2018-01-04 14:56:19   
46869    delivered      2017-06-10 23:12:16  2017-06-10 23:35:12   
91077    delivered      2018-08-20 00:55:24  2018-08-21 04:50:24   
77800    delivered      2017-01-23 22:55:07  2017-01-23 23:10:11   
15320    delivered      2018-01-06 20:03:11  2018-01-06 20:17:25   
31283    delivered      2017-11-08 17:07:41  2017-11-09 04:31:09   
29704    delivered      2018-07-28 19:17:11  2018-07-28 19:30:33   
59310    delivered      2018-03-09 18:04:09  2018-03-09 19:08:29   
75314    delivered      2017-05-20 09:16:51  2017-05-20 09:31:35   

      order_delivered_carrier_date order_delivered_customer_date  \
51803          2017-06-09 14:16:00           2017-06-13 11:39:49   
49025          2018-01-16 20:28:47           2018-01-30 21:21:55   
46869          2017-06-12 09:33:58           2017-07-03 13:03:28   
91077          2018-08-22 13:12:00           2018-08-23 20:41:04   
77800          2017-01-24 15:05:01           2017-01-30 12:13:46   
15320          2018-01-10 18:09:48           2018-01-22 23:18:57   
31283          2017-11-10 21:21:15           2017-11-13 18:12:45   
29704          2018-07-30 08:43:00           2018-08-07 16:41:39   
59310          2018-03-12 14:51:40           2018-03-21 18:20:37   
75314          2017-05-25 16:15:50           2017-06-05 17:53:40   

      order_estimated_delivery_date  
51803           2017-06-22 00:00:00  
49025           2018-01-29 00:00:00  
46869           2017-07-17 00:00:00  
91077           2018-08-27 00:00:00  
77800           2017-02-20 00:00:00  
15320           2018-02-08 00:00:00  
31283           2017-12-06 00:00:00  
29704           2018-08-08 00:00:00  
59310           2018-05-04 00:00:00  
75314           2017-06-16 00:00:00

In [76]:
# mombres de columnas
Orders.columns

Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date'],
      dtype='object')

In [85]:
# tamaño del dataset
Orders.shape

(99441, 8)

In [42]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB


In [95]:
# descripcion de las columnas numéricas
Orders.describe()

order_id                       customer_id  \
count                              99441                             99441   
unique                             99441                             99441   
top     e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
freq                                   1                                 1   

       order_status order_purchase_timestamp    order_approved_at  \
count         99441                    99441                99281   
unique            8                    98875                90733   
top       delivered      2018-04-11 10:48:14  2018-02-27 04:31:10   
freq          96478                        3                    9   

       order_delivered_carrier_date order_delivered_customer_date  \
count                         97658                         96476   
unique                        81018                         95664   
top             2018-05-09 15:48:00           2018-05-08 23:38:46   
freq                             47                             3   

       order_estimated_delivery_date  
count                          99441  
unique                           459  
top              2017-12-20 00:00:00  
freq                             522

In [96]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Orders.describe(include=['O'])

order_id                       customer_id  \
count                              99441                             99441   
unique                             99441                             99441   
top     e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
freq                                   1                                 1   

       order_status order_purchase_timestamp    order_approved_at  \
count         99441                    99441                99281   
unique            8                    98875                90733   
top       delivered      2018-04-11 10:48:14  2018-02-27 04:31:10   
freq          96478                        3                    9   

       order_delivered_carrier_date order_delivered_customer_date  \
count                         97658                         96476   
unique                        81018                         95664   
top             2018-05-09 15:48:00           2018-05-08 23:38:46   
freq                             47                             3   

       order_estimated_delivery_date  
count                          99441  
unique                           459  
top              2017-12-20 00:00:00  
freq                             522

In [44]:
# cuantos datos nulos por columna se encuentran
Orders.isna().sum()

order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

## Products

In [46]:
# carga del dataset y muestreo de 10 registros aleatorios
Products = pd.read_csv('Datasets/olist_products_dataset.csv')
Products.sample(10)

product_id  product_category_name  \
25387  626fcce994b573d0bd54d6447d5badd3                    NaN   
4812   cda185193c72c0eb8dde03f20f101230  utilidades_domesticas   
3560   94e87c8d2cac9ca6fa8c91b99712caaf         telefonia_fixa   
1776   b22934d8dedd07b3033be2bf6da70bcb             brinquedos   
18224  9eff8192b50cf828d20417ae6c41f4b5           beleza_saude   
24930  a1d54bd2293d947527d70e04dbe2a93e           climatizacao   
29594  1046440f587e8b22c047de8ab39095dc                  bebes   
20770  7d761cef314675473eda87ecc3b7e751        cama_mesa_banho   
11466  1b5b6bd531d2e0b1650bbabac62fa352  utilidades_domesticas   
988    9d358fd2a77935cffa32834b50cfb21e             cool_stuff   

       product_name_lenght  product_description_lenght  product_photos_qty  \
25387                  NaN                         NaN                 NaN   
4812                  57.0                      1782.0                 1.0   
3560                  47.0                       178.0                 2.0   
1776                  33.0                       561.0                 2.0   
18224                 26.0                      1076.0                 1.0   
24930                 45.0                       638.0                 1.0   
29594                 59.0                       832.0                 4.0   
20770                 45.0                       729.0                 1.0   
11466                 50.0                       364.0                 1.0   
988                   53.0                       953.0                 1.0   

       product_weight_g  product_length_cm  product_height_cm  \
25387             300.0               18.0                6.0   
4812             1100.0               24.0               24.0   
3560              675.0               20.0               23.0   
1776              415.0               19.0               16.0   
18224            2900.0               21.0               16.0   
24930           19275.0               85.0               42.0   
29594             650.0               38.0               15.0   
20770            1250.0               20.0               30.0   
11466            7900.0               62.0               28.0   
988              1200.0               42.0               25.0   

       product_width_cm  
25387              11.0  
4812               24.0  
3560               19.0  
1776               17.0  
18224              31.0  
24930              40.0  
29594              43.0  
20770              30.0  
11466              28.0  
988                15.0

In [77]:
# mombres de columnas
Products.columns

Index(['product_id', 'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm'],
      dtype='object')

In [86]:
# tamaño del dataset
Products.shape

(32951, 9)

In [47]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB


In [89]:
# descripcion de las columnas numéricas
Products.describe()

product_name_lenght  product_description_lenght  product_photos_qty  \
count         32341.000000                32341.000000        32341.000000   
mean             48.476949                  771.495285            2.188986   
std              10.245741                  635.115225            1.736766   
min               5.000000                    4.000000            1.000000   
25%              42.000000                  339.000000            1.000000   
50%              51.000000                  595.000000            1.000000   
75%              57.000000                  972.000000            3.000000   
max              76.000000                 3992.000000           20.000000   

       product_weight_g  product_length_cm  product_height_cm  \
count      32949.000000       32949.000000       32949.000000   
mean        2276.472488          30.815078          16.937661   
std         4282.038731          16.914458          13.637554   
min            0.000000           7.000000           2.000000   
25%          300.000000          18.000000           8.000000   
50%          700.000000          25.000000          13.000000   
75%         1900.000000          38.000000          21.000000   
max        40425.000000         105.000000         105.000000   

       product_width_cm  
count      32949.000000  
mean          23.196728  
std           12.079047  
min            6.000000  
25%           15.000000  
50%           20.000000  
75%           30.000000  
max          118.000000

In [97]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Products.describe(include=['O'])

product_id product_category_name
count                              32951                 32341
unique                             32951                    73
top     1e9e8ef04dbcff4541ed26657ea517e5       cama_mesa_banho
freq                                   1                  3029

In [49]:
# cuantos datos nulos por columna se encuentran
Products.isna().sum()

product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

## Sellers

In [51]:
# carga del dataset y muestreo de 10 registros aleatorios
Sellers = pd.read_csv('Datasets/olist_sellers_dataset.csv')
Sellers.sample(10)

seller_id  seller_zip_code_prefix  \
272   f0ec6a2adb05c62655a26dd347b8dede                    1303   
423   ceaec5548eefc6e23e6607c5435102e7                    3821   
24    8bd0e3abda539b9479c4b44a691be1ec                   98910   
2495  dfc475d54e1b6dbeeb7d7d9bdaa63827                   81460   
628   38102b031c2a15e54623d711bfc753d3                   62170   
2755  a4fc6eadaff1576af888509a53420f6a                    9561   
2784  ce616e1913288884e7742faac9d981db                    6711   
457   714b9a1332840f257b49f476bea4e341                   15910   
749   05feb94f19d094d4b0f9281f0b1d4c99                    8752   
722   9dd59e43f0c9f0553244e8f9c2fb247e                   85851   

             seller_city seller_state  
272            sao paulo           SP  
423            sao paulo           SP  
24          tres de maio           RS  
2495            curitiba           PR  
628              mucambo           CE  
2755  sao caetano do sul           SP  
2784               cotia           SP  
457           monte alto           SP  
749      mogi das cruzes           SP  
722        foz do iguacu           PR

In [78]:
# mombres de columnas
Sellers.columns

Index(['seller_id', 'seller_zip_code_prefix', 'seller_city', 'seller_state'], dtype='object')

In [87]:
# tamaño del dataset
Sellers.shape

(3095, 4)

In [52]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Sellers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   seller_id               3095 non-null   object
 1   seller_zip_code_prefix  3095 non-null   int64 
 2   seller_city             3095 non-null   object
 3   seller_state            3095 non-null   object
dtypes: int64(1), object(3)
memory usage: 96.8+ KB


In [90]:
# descripcion de las columnas numéricas
Sellers.describe()

seller_zip_code_prefix
count             3095.000000
mean             32291.059451
std              32713.453830
min               1001.000000
25%               7093.500000
50%              14940.000000
75%              64552.500000
max              99730.000000

In [98]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Sellers.describe(include=['O'])

seller_id seller_city seller_state
count                               3095        3095         3095
unique                              3095         611           23
top     3442f8959a84dea7ee197c632cb2df15   sao paulo           SP
freq                                   1         694         1849

In [53]:
# cuantos datos nulos por columna se encuentran
Sellers.isna().sum()

seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64

## Product_category_name_translation

In [55]:
# carga del dataset y muestreo de 10 registros aleatorios
Product_category_name_translation = pd.read_csv('Datasets/product_category_name_translation.csv')
Product_category_name_translation.sample(10)

product_category_name product_category_name_english
70        seguros_e_servicos         security_and_services
33          eletrodomesticos               home_appliances
13  tablets_impressao_imagem        tablets_printing_image
1     informatica_acessorios         computers_accessories
51           fashion_esporte                 fashion_sport
35             casa_conforto                  home_confort
53                       pcs                     computers
2                 automotivo                          auto
57         livros_importados                books_imported
9         relogios_presentes                 watches_gifts

In [79]:
# mombres de columnas
Product_category_name_translation.columns

Index(['product_category_name', 'product_category_name_english'], dtype='object')

In [88]:
# tamaño del dataset
Product_category_name_translation.shape

(71, 2)

In [56]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Product_category_name_translation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   product_category_name          71 non-null     object
 1   product_category_name_english  71 non-null     object
dtypes: object(2)
memory usage: 1.2+ KB


In [57]:
# descripcion de las columnas numéricas
Product_category_name_translation.describe()

product_category_name product_category_name_english
count                     71                            71
unique                    71                            71
top             beleza_saude                 health_beauty
freq                       1                             1

In [99]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Product_category_name_translation.describe(include=['O'])

product_category_name product_category_name_english
count                     71                            71
unique                    71                            71
top             beleza_saude                 health_beauty
freq                       1                             1

In [58]:
# cuantos datos nulos por columna se encuentran
Product_category_name_translation.isna().sum()

product_category_name            0
product_category_name_english    0
dtype: int64